src/notebooks/controllers/commands/serverConnectionControllerCommands.ts

```diff
@@ -0,0 +1,307 @@
+        const multiStep = this.multiStepFactory.create<{}>();
+        return multiStep.run(
+            this.startSwitchRun.bind(
+                this,
+                this.startSwitchingToLocal.bind(this, activeNotebookType, startingLocal, startingUri)
+            ),
+            {}
+        );
+    }
+
+    private async startSwitchingToLocal(
+        activeNotebookType: typeof JupyterNotebookView | typeof InteractiveWindowView,
+        startedLocal: boolean,
+        startedUri: string | undefined,
+        input: IMultiStepInput<{}>
+    ): Promise<InputStep<{}> | void> {
+        // Wait until we switch to local
+        const deferred = createDeferred<boolean>();
+        const disposable = this.serverConnectionType.onDidChange(() => deferred.resolve(true));
+        try {
+            await this.serverSelector.setJupyterURIToLocal();
+            await deferred.promise;
+        } finally {
+            disposable.dispose();
+        }
+        try {
+            // Then bring up the quick pick for the current set of controllers.
+            await this.showKernelPicker(
+                DataScience.pickLocalKernelTitle(),
+                DataScience.pickLocalKernelPlaceholder(),
+                true,
+                activeNotebookType,
+                input
+            );
+        } catch (e) {
+            // They backed out. Put back to local
+            if (!startedLocal && startedUri) {
+                await this.serverSelector.setJupyterURIToRemote(startedUri, true);
+            }
+            if (e === InputFlowAction.back) {
+                return this.showVsCodeKernelPicker();
+            }
+        }
+    }
+
+    private async showKernelPicker(
+        title: string,
+        placeholder: string,
+        local: boolean,
+        viewType: typeof JupyterNotebookView | typeof InteractiveWindowView,
+        input: IMultiStepInput<{}>
+    ): Promise<void> {
+        // Get the current list. We will dynamically update the list as
+        // more and more controllers are found.
+        const controllers = this.controllerRegistration.values.filter(
+            (c) => isLocalConnection(c.connection) === local && c.viewType === viewType
+        );
+
+        // Create an event emitter for when new controllers are added
+        const changeEmitter = new EventEmitter<(QuickPickItem | ControllerQuickPick)[]>();
+
+        // Use the current controllers to generate the quick pick items
+        const picks: ControllerQuickPick[] = controllers.map((d) => {
+            return {
+                label: d.label,
+                detail: undefined,
+                description: d.controller.description,
+                controller: d
+            };
+        });
+
+        // Then group them by kind
+        const kernelsPerCategory = groupBy(picks, (a, b) =>
+            compareIgnoreCase(a.controller.controller.kind || 'z', b.controller.controller.kind || 'z')
```

Comment:
Is the 'z' for alphabetizing? Doesn't seem like that would work for non-en.

Reply:
I copied this from the core, but it does seem wrong. 'z' only works in ascii.

Reply:
I'll enter an issue to relook at this in both the core and here.